File: Lab_3/cm3038-python-lab03/Python/aips/lab03/test/testSuccessor.py
import aips.lab03.sq8 as sq8


"""Test successor function of a state.
"""
def run():
    num1=[[8,7,6],[5,4,3],[2,1,0]]
    s=sq8.Sq8State(num1)

    print("State:\n{}".format(s))
    print("All action-state:\n")
    for actionState in s.successor():
        print("{}{}".format(actionState.action,actionState.state))

    num2=[[8,7,6],[5,0,4],[3,2,1]]
    s=sq8.Sq8State(num2)

    print("State:\n{}".format(s))
    print("All action-state:\n")
    for actionState in s.successor():
        print("{}{}".format(actionState.action,actionState.state))

if __name__=="__main__":
    run()

File: cm3038-Java-lab06/Java/aips/lab06/tictactoe/test/TestSuccessor.java
package aips.lab06.tictactoe.test;

import aips.lab06.tictactoe.*;
import aips.search.adversarial.PlayerRole;

public class TestSuccessor {
public static void main(String[] arg)
{
int[][] symbol={	{1,0,2},
				{0,0,0},
				{2,0,1},
				};

TttState s=new TttState(symbol);		//create game state
System.out.println("State:\n"+s.toString());
System.out.println("Successors for MAX player:\n");
System.out.println("All action-state:\n"+s.successor(PlayerRole.MAX).toString());

System.out.println("Successors for MIN player:\n");
System.out.println("All action-state:\n"+s.successor(PlayerRole.MIN).toString());
} //end method
} //end class